### ISSUE TITLE
Failed to compile polyval on GPU in Linalg on tensors

### ISSUE THREAD

**Issue by hanhanW** [google/iree#5311]
**Describe the bug**
This is derived from python integration tests. It failed because of hitting an assertion: `workloadPerWorkgroup.size() <= 3 && "workloadPerWorkgroup size greater than 3 not handled"`

Full log with `print-ir-after-all`: https://gist.github.com/hanhanW/e6379ccb49734d02fbec72a812898a09

**To Reproduce**

```mlir
module  {
  func @polyval__three_coeffs(%arg0: tensor<2x2x2x2x2x2x2xf32> {tf._user_specified_name = "args_0"}, %arg1: tensor<2x2x2x2x2x2x2xf32> {tf._user_specified_name = "args_0"}, %arg2: tensor<2x2x2x2x2x2x2xf32> {tf._user_specified_name = "args_0"}, %arg3: tensor<f32> {tf._user_specified_name = "args_1"}) -> tensor<2x2x2x2x2x2x2xf32> attributes {iree.module.export, iree.reflection = {abi = "sip", abiv = 1 : i32, sip = "I27!S23!k0S13!k0_0k1_1k2_2k1_3R3!_0"}, tf._construction_context = "kEagerRuntime", tf._input_shapes = [#tf.shape<2x2x2x2x2x2x2>, #tf.shape<2x2x2x2x2x2x2>, #tf.shape<2x2x2x2x2x2x2>, #tf.shape<>]} {
    %0 = "mhlo.broadcast_in_dim"(%arg3) {broadcast_dimensions = dense<> : tensor<0xi64>} : (tensor<f32>) -> tensor<2x2x2x2x2x2x2xf32>
    %1 = mhlo.multiply %arg0, %0 : tensor<2x2x2x2x2x2x2xf32>
    %2 = mhlo.add %arg1, %1 : tensor<2x2x2x2x2x2x2xf32>
    %3 = "mhlo.broadcast_in_dim"(%arg3) {broadcast_dimensions = dense<> : tensor<0xi64>} : (tensor<f32>) -> tensor<2x2x2x2x2x2x2xf32>
    %4 = mhlo.multiply %2, %3 : tensor<2x2x2x2x2x2x2xf32>
    %5 = mhlo.add %arg2, %4 : tensor<2x2x2x2x2x2x2xf32>
    return %5 : tensor<2x2x2x2x2x2x2xf32>
  }
}
```

Run `iree-translate --iree-mlir-to-vm-bytecode-module --iree-hal-target-backends=vulkan-spirv -iree-flow-dispatch-linalg-on-tensors=true -iree-codegen-spirv-experimental-linalg-on-tensors=true polyval.mlir`

**Comment by MaheshRavishankar** [comment]
Will see if PR #5297 will help there

**Comment by MaheshRavishankar** [comment]
THis is fixed by PR #5297 (Gist after all : https://gist.github.com/MaheshRavishankar/dba215aed6dbbf167afa03c9d8631305)
